lib/src/main/java/com/cloudbees/groovy/cps/impl/MethodPointerBlock.java

```diff
@@ -64,6 +65,9 @@ public Next fixLhs(Object lhs) {
          * Obtain a method pointer, which is really just a {@link MethodClosure}.
          */
         public Next done(Object methodName) {
+            if (methodName instanceof GString) {
+                methodName = methodName.toString();
```

Comment:
Can this call something outside the sandbox?

Reply:
No, this is the same as code like this in `DSL.java`: https://github.com/jenkinsci/workflow-cps-plugin/blob/a8fc49e2f96fbb9aa92860b6fa053c70734f0452/plugin/src/main/java/org/jenkinsci/plugins/workflow/cps/DSL.java#L601-L602

It can call closures embedded in the `GString` (simple expressions have already been fully evaluated by this point), but any closures it calls will have been CPS and sandbox-transformed.

Reply:
Also though, the behavior of `GString`-embedded closures in this context is totally broken (regardless of this patch), even worse than described [here](https://www.jenkins.io/doc/book/pipeline/cps-method-mismatches/#PipelineCPSmethodmismatches-ClosuresinsideGString) (somehow the argument to the closure is passed as an extra argument to `new GStringImpl`). Maybe we should update `visitGString` to just fail immediately if it contains a `ClosureExpression`.

That said, this PR is not really important, it just looked like it might be easy to fix.